python/prophet/forecaster.py

```diff
@@ -1438,19 +1482,160 @@ def predict_uncertainty(self, df):
         -------
         Dataframe with uncertainty intervals.
         """
-        sim_values = self.sample_posterior_predictive(df)
+        # use vectorization only if mcmc_sampling is not used
+        if self.mcmc_samples > 0:
+            sim_values = self.sample_posterior_predictive(df)
```

Comment:
again I'm not sure I undertstand - no mcmc_samples means not creating an uncertainty interval doesn't it? why are we creating a single sample and then calculating the percentiles on it?

Reply:
This is only being evaluated if mcmc_sampling is being used. This section here is just the original code, unmodified.